fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/expressions/LiteralKeyExpressionTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * LiteralKeyExpressionTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.metadata.expressions;
+
+import com.apple.foundationdb.record.metadata.Key;
+import com.apple.foundationdb.record.provider.common.text.TextSamples;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.annotation.Nonnull;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+/**
+ * Unit tests for the {@link LiteralKeyExpression}.
+ */
+public class LiteralKeyExpressionTest {
+    private static final byte[] byteArray = {0x00, 0x0f, 0x16};
+
+    private static Stream<Object> correctValues() {
+        return Stream.of(
+                1.0f,
+                2.0d,
+                100L,
+                Long.MIN_VALUE,
+                Long.MAX_VALUE,
+                true,
+                false,
+                "a string",
+                "(╯°□°)╯︵ ┻━┻",
+                "┳━┳ ヽ(ಠل͜ಠ)ﾉ",
```

Comment:
So there's a weird other case that might be worth seeing what it does:

`"malformed surrogate pair: \uD83C"`

That unicode character is a high-surrogate and it's missing its low surrogate buddy, so it can't be expressed in UTF-8, so Protobuf might do weird things with it.

Reply:
Added a test. It works fine if we do everything with Java's protobuf objects, but things break if we add a byte serialization in the middle.